### PR TITLE
test java 11

### PR DESCRIPTION
**Pull request by danielnorberg**
# Hey, I just made a Pull Request!

## Description
Ensure that Styx works on the newly released Java 11.

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [x] All tests pass
- [x] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [x] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

**Comment by danielnorberg** [comment]
```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /root/styx/styx-api-service/src/test/java/com/spotify/styx/api/workflow/WorkflowInitializerTest.java:[68,33] error: unreported exception Exception; must be caught or declared to be thrown
[INFO] 1 error
```

But afaict no method called there throws `Exception`.

**Comment by danielnorberg** [comment]
Oh, I guess it's `runInTransaction` which `throws IOException, E`. And type inference rules has changed.

File: styx-api-service/src/test/java/com/spotify/styx/api/workflow/WorkflowInitializerTest.java [hunk 1]
```diff
@@ -63,7 +63,7 @@
   @Mock private StorageTransaction transaction;
 
   @Before
-  public void setUp() throws IOException {
+  public void setUp() throws Exception {
```
Comment (honnix) [inline styx-api-service/src/test/java/com/spotify/styx/api/workflow/WorkflowInitializerTest.java]: What caused this change?

Comment (danielnorberg) [reply styx-api-service/src/test/java/com/spotify/styx/api/workflow/WorkflowInitializerTest.java]: Javac bug in JDK 11 

https://github.com/danielnorberg/jdk11-behavior-change-repro

Comment (honnix) [reply styx-api-service/src/test/java/com/spotify/styx/api/workflow/WorkflowInitializerTest.java]: Awesome!

File: .circleci/config.yml [hunk 2]
```diff
@@ -37,17 +37,17 @@ jobs:
           when: always
           command: mvn -B clean compile test-compile
 
-  test_jdk10:
+  test_jdk11:
```
[the remark this answers is not in the document]
Comment (danielnorberg) [reply .circleci/config.yml]: Yeah, don't want duplicate reports send to codecov. 

I do want to test and coverage reports for both jdk8 and jdk11 builds in circleci though.

[the remark this answers is not in the document]
Comment (honnix) [reply .circleci/config.yml]: nvm, i was concerning about naming only.

**Comment by honnix** [comment]
👍

**Pull request merged**